extensions/ql-vscode/src/queries-panel/query-tree-view-item.ts

```diff
@@ -24,3 +30,9 @@ export class QueryTreeViewItem extends vscode.TreeItem {
     }
   }
 }
+
+export class QueryTreeTextItem extends QueryTreeViewItem {
+  constructor(text: string) {
+    super(text);
+  }
+}
```

Comment:
Why is this type necessary? We can make the path property of the old `QueryTreeViewItem` optional and reuse that type?

Reply:
The reason for making subclasses is precisely so that the `path` property can remain non-optional.

The idea I'm going for is that as far as is practical it should be impossible to construct a class an in invalid state. If we kept it all as one class with all fields marked as optional, it would be possible to construct an instance where fields were undefined but values are expected by other areas of code. For example, code like https://github.com/github/vscode-codeql/blob/d290b56649527d55695acae73d5df3abb5b7a35c/extensions/ql-vscode/src/local-queries/local-queries.ts#L146 expects `path` to be defined for every node.

Also, if `QueryTreeTextItem` did have more fields or any implementation details, that complexity would not affect `QueryTreeQueryItem`. We keep code separate so each bit implements only what it needs, instead of having one class that tries to cover every case.

Reply:
I see that [`DbTreeViewItem`](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/databases/ui/db-tree-view-item.ts) addresses this same concern in a different way.

There, there is one class that covers all cases, so over half of the fields are optional. However, the class constructor is not used outside of that file and there are a large number of exported `createDbTreeViewItemXXX` methods that cover the different cases.

That solves the problem of creating valid instances since stricter typing can be used on the exported methods. It doesn't solve the problem when reading data since then you'll always dealing with a `DbTreeViewItem` so you have to assume data is valid and fields are defined when they should be. However it does avoid the added complexity of having more classes and inheritance.

Overall, I don't mind going for that strategy instead. I think either works and it's not clear that one is strictly better than the other, but being consistent is nice.

Reply:
I think consistency is important. I've converted it to match the style from the queries panel using exported functions to create `QueryTreeViewItem`s.